jni/generators/synthesizer.cpp

```diff
@@ -312,10 +308,8 @@ void Synthesizer::render( AudioBuffer* aOutputBuffer, BaseSynthEvent* aEvent )
 
     if ( !hasParent )
     {
-        _instrument->adsr->setLastEnvelope( aEvent->cachedProps.ADSRenvelope );
-        _instrument->adsr->setBufferLength( aEvent->getEventLength() );
+        _instrument->adsr->apply( aOutputBuffer, aEvent, bufferWriteIndex );
 
```

Comment:
Let the ADSR module manage all cached envelope values and buffer lengths, Synthesizer shouldn't care.